tests/unit/Option/NoneTest.php

```diff
@@ -84,6 +84,26 @@ public function testContains(): void
         static::assertFalse($option->contains(4));
     }
 
+    public function testProceed(): void
+    {
+        $checked_divisor = static function (int $dividend, int $divisor): Option\Option {
```

Comment:
Just wondering : do we need these large tests? 
The things you put in here seem rather like examples.

Maybe it could make sense to just do something like `$actual = none()->proceed(...)` and verify the correct return? That way, it makes sure there is no logic issue in the example and on top of that: one does not need to grasp the example first before seeing what is wrong with the test-case.

For other components, we are adding README.md files to the component root with examples and more information about the way it works. Maybe it makes more sense to go that way?

(for example: https://github.com/azjezz/psl/tree/next/src/Psl/Async#readme)

Reply:
Indeed I've taken the examples from the Rust documentation to be familiar, but I agree they would better be in a markdown file.

I'm not confident in my English and I also don't know in depth a lot of components, so I'm somewhat scared of writing documentation. I guess I can try it in another pull request and get corrected in the reviews, but I don't want to spam a lot of them too 😊